cdap-hbase-compat-0.98/src/main/java/co/cask/cdap/data2/transaction/messaging/coprocessor/hbase98/MessageTableRegionObserver.java

```diff
@@ -96,16 +97,20 @@ public void start(CoprocessorEnvironment env) throws IOException {
     if (env instanceof RegionCoprocessorEnvironment) {
       HTableDescriptor tableDesc = ((RegionCoprocessorEnvironment) env).getRegion().getTableDesc();
       String metadataTableNamespace = tableDesc.getValue(Constants.MessagingSystem.HBASE_METADATA_TABLE_NAMESPACE);
-      String metadataTableName = tableDesc.getValue(Constants.MessagingSystem.HBASE_METADATA_TABLE_NAME);
       String hbaseNamespacePrefix = tableDesc.getValue(Constants.Dataset.TABLE_PREFIX);
+
+      HTableNameConverter nameConverter = new HTable98NameConverter();
+      String sysConfigTablePrefix = nameConverter.getSysConfigTablePrefix(hbaseNamespacePrefix);
+      cConfReader = new CConfigurationReader(env.getConfiguration(), sysConfigTablePrefix);
+      CConfiguration cConf = cConfReader.read();
+      String metadataTableName = cConf.get(Constants.MessagingSystem.METADATA_TABLE_NAME);
+
       prefixLength = Integer.valueOf(tableDesc.getValue(
         Constants.MessagingSystem.HBASE_MESSAGING_TABLE_PREFIX_NUM_BYTES));
-      HTableNameConverter nameConverter = new HTable98NameConverter();
+
       metadataTable = env.getTable(nameConverter.toTableName(hbaseNamespacePrefix,
                                                              TableId.from(metadataTableNamespace, metadataTableName)));
-      String sysConfigTablePrefix = nameConverter.getSysConfigTablePrefix(hbaseNamespacePrefix);
-      cConfReader = new CConfigurationReader(env.getConfiguration(), sysConfigTablePrefix);
-      final Long metadataCacheExpiry = Long.parseLong(cConfReader.read().get(
+      final Long metadataCacheExpiry = Long.parseLong(cConf.get(
```

Comment:
`cConf` has a `getLong` method that will do the parsing. Also, the type should be `long`, not `Long`